src/toggle.jsx

```diff
@@ -216,39 +223,39 @@ const Toggle = React.createClass({
       thumbStyles.marginLeft = '-' + thumbStyles.width;
     }
 
-    let toggleElementStyles = this.mergeStyles(styles.toggleElement, this.props.elementStyle);
+    const toggleElementStyles = Object.assign({},
+      styles.toggleElement,
+      this.props.elementStyle
+    );
 
-    let toggleElement = (
-      <div style={this.prepareStyles(toggleElementStyles)}>
-        <div style={this.prepareStyles(trackStyles)} />
+    const toggleElement = (
+      <div style={prepareStyles(Object.assign({}, toggleElementStyles))}>
+        <div style={prepareStyles(Object.assign({}, trackStyles))} />
         <Paper style={thumbStyles} circle={true} zDepth={1} />
       </div>
     );
 
-    let customRippleStyle = this.mergeStyles({
-      top: -10,
-      left: -10,
-    }, this.props.rippleStyle);
-
-    let rippleColor = this.state.switched ?
-      this.getTheme().thumbOnColor : this.state.muiTheme.textColor;
+    const rippleStyle = Object.assign({},
+      styles.ripple,
+      this.props.rippleStyle
+    );
 
-    let iconStyle = this.mergeStyles(
+    const iconStyle = Object.assign({},
```

Comment:
Yeah I think I noticed this too when I was doing it. If I remember correctly it was `iconButtonElement` that could possibly be reassigned. If we pass the elements that get passed in as props to a stateless internal component, that could handle the logic and we could probably make all of these `const` then.

Though that process would be tough to automate.